core/app/models/spree/return_authorization.rb

```diff
@@ -20,6 +21,8 @@ class ReturnAuthorization < Spree::Base
     validate :must_have_shipped_units, on: :create
     validate :no_previously_exchanged_inventory_units, on: :create
 
+    validates :recipient, presence: true, if: :override_recipient
+    attr_accessor :override_recipient
```

Comment:
@jhawthorn We thought of a couple options here. 

* Override `create` in 
https://github.com/solidusio/solidus/blob/master/backend/app/controllers/spree/admin/return_authorizations_controller.rb which inherits from https://github.com/solidusio/solidus/blob/master/backend/app/controllers/spree/admin/resource_controller.rb#L52. We could allow `create` to be aware of the user wanting to override a recipient.
* The current approach, which hooks into the AM attributes hooks.

Are you thinking of another option or do you prefer to leave the logic in the controller? We went with the model level approach because 1) we don't have to override `#create` 2) We get validation messages for free. For example, if a user checks the `Override Recipient` box and fails to select a recipient, we automatically get the `Recipient can't be blank` error message.